### ISSUE TITLE
Builder error. Unable to compile when `if` statement, which contains an `or` block, is used as an expression

### ISSUE THREAD

**Issue by kukaki** [vlang/v#11273]
**V version:** V 0.2.2 da53f81.dd486bb
**OS:** linux, Ubuntu 21.04

**What did you do?**
```
$ v run .
```

**What did you expect to see?**
Normal compilation, and execution, if no errors.
 
**What did you see instead?**
```
/tmp/v/backend.15431756830905292250.tmp.c:22976: warning: assignment makes pointer from integer without a cast
/tmp/v/backend.15431756830905292250.tmp.c:22990: warning: assignment makes pointer from integer without a cast
/tmp/v/backend.15431756830905292250.tmp.c:30925: error: ')' expected (got "_t8")
builder error: 
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please report the bug using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

**V is unable to compile the following code.**

```v
tpos_rank := if r.fpos_id == r.tpos_id { fpos_rank } else { app.get_position_rank_by_id(r.tpos_id) or {
				return true
			} }
```

**Comment by JalonSolov** [comment]
Try changing the `or` block to
```v
or { true }
```
otherwise, we may need a more complete example.

In the meantime, yes, V definitely needs to catch this and give a better error message.

**Comment by kukaki** [comment]
> otherwise, we may need a more complete example.

The following works.
```v
fn (mut app App) is_requirement_invalid(rqt Requirement) bool {
  fpos_rank := app.get_position_rank_by_id(rqt.fpos_id) or { return true }
  mut tpos_rank := fpos_rank
  if rqt.fpos_id != rqt.tpos_id {
    tpos_rank = app.get_position_rank_by_id(rqt.tpos_id) or { return true }
  }
 // ...
  return false
}
```
While the following results in builder error.
```v
fn (mut app App) is_requirement_invalid(rqt Requirement) bool {
  fpos_rank := app.get_position_rank_by_id(rqt.fpos_id) or { return true }
  tpos_rank := if rqt.fpos_id == rqt.tpos_id { fpos_rank } else {
    app.get_position_rank_by_id(rqt.tpos_id) or {  return true  }
  }
 // ...
  return false
}
```

**Comment by spytheman** [comment]
Fixed in cc4af23 .
